### PR TITLE
[ENG-1628] Move Egap registirations to EGAP provider

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

We need to set all Registrations with Egap schemas to thier EGAP provider.

## Changes

- add simple management command and test.

## QA Notes

Should be good with just unit test and dry run.

## Documentation

🐞 fix, no docs.

## Ticket

https://openscience.atlassian.net/browse/ENG-1628

### CODE REVIEW

File: osf_tests/management_commands/test_move_egap_regs_to_provider.py [hunk 1]
```diff
@@ -34,7 +38,7 @@ def egap_reg(self):
         cos = RegistrationProviderFactory(_id='osf')
         return RegistrationFactory(schema=egap_schema, provider=cos)
 
-    def test_move_to_provider(self, egap_provider, egap_reg):
+    def test_move_to_provider(self, egap_provider, egap_reg, non_egap_provider):
         assert egap_reg.provider != egap_provider
```
Comment (brianjgeiger) [inline osf_tests/management_commands/test_move_egap_regs_to_provider.py:43]: You didn't seem to add any assertions about the non_egap_provider to ensure that it was unchanged after the script ran.

Comment (Johnetordoff) [reply osf_tests/management_commands/test_move_egap_regs_to_provider.py:43]: Ah I see now.

Comment (Johnetordoff) [reply osf_tests/management_commands/test_move_egap_regs_to_provider.py:48]: ok added here https://github.com/CenterForOpenScience/osf.io/pull/9351/files#diff-5f90a7ad30edf0f466a1c019f7bd9692R53

File: osf/management/commands/move_egap_regs_to_provider.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+def main(dry_run):
+    epag_provider = RegistrationProvider.objects.get(name=settings.EGAP_PROVIDER_NAME)
+
+    egap_schema = RegistrationSchema.objects.filter(
+        name='EGAP Registration'
+    ).order_by(
+        '-schema_version'
+    )[0]
+
+    egap_regs = Registration.objects.filter(registered_schema=egap_schema.id, provider___id='osf')
```
Comment (mfraezz) [inline osf/management/commands/move_egap_regs_to_provider.py:26]: Minor: `.exclude(provider___id='egap')` would technically be more correct, but as there aren't currently any non-`osf` `RegistrationProvider`'s, it's probably fine.

File: osf/management/commands/move_egap_regs_to_provider.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    RegistrationSchema,
+    Registration
+)
+from django.conf import settings
+
+def main(dry_run):
+    epag_provider = RegistrationProvider.objects.get(name=settings.EGAP_PROVIDER_NAME)
+
+    egap_schema = RegistrationSchema.objects.filter(
+        name='EGAP Registration'
```
Comment (brianjgeiger) [inline osf/management/commands/move_egap_regs_to_provider.py:21]: There are now two EGAP Registration schemas, so we'll have to account for that in this script.

**Pull request merged**